Video Processor bug using the v4l2loopback device

## Description

After selecting Start Video Processor with an active video stream to /dev/video0 using ffmpeg's x11grab functionality (verifiable by using ffplay and mpv to get video from the /dev/video0 device), RetroArch exits out, stating that "Libretro core requires content, but nothing was provided." Given that ffplay and mpv can handle the output, this leads me to believe that it is an issue in either the ffmpeg core (also as a side issue it has issues with .webm videos generated by yt-dlp but that's not this issue) or it's in the v4l2 code itself.

### Expected behavior

RetroArch would then load the video stream and allow me to do what I want to do with it, which is to apply live post-processing shaders.

### Actual behavior

The v4l2 core simply errors out.

### Steps to reproduce the bug

1. Load the v4l2loopback device with `sudo modprobe v4l2loopback`
2. Start the video stream with `ffmpeg -f x11grab -r 60 -s 1920x1080 -i :0.0+0,0 -vcodec rawvideo -f v4l2 /dev/video0`
3. Load RetroArch and go to the Start Video Processor option in Load Core.


### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.14.0/ad89b0c655

### Environment information

- OS: Arch Linux 6.1.9-arch1-1
- Compiler: Irrelevant, using repository's package

### Relevant Logs
**Before Launching the v4l2 core**
```
[INFO] RetroArch 1.14.0 (Git ad89b0c655)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
[INFO] Capabilities: MMX MMXEXT SSE SSE2 SSE3 SSE4 SSE4.2 AES AVX AVX2
[INFO] Built: Dec 13 2022
[INFO] Version: 1.14.0
[INFO] Git: ad89b0c655
[INFO] =================================================
[INFO] [Input]: Found input driver: "x".
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Set video size to: 960x720.
[ERROR] [Wayland]: Failed to connect to Wayland server.
[INFO] [GLX]: GLX_EXT_swap_control_tear supported.
[INFO] [GLCore]: Found GL context: "x".
[INFO] [GLCore]: Detecting screen resolution: 1920x1080.
[INFO] [GLX]: X = 0, Y = 0, W = 960, H = 720.
[INFO] [GLX]: Creating context for requested version 3.2.
[INFO] [GLX]: Not running Mesa, trying higher versions...
[INFO] [GLX]: Creating context for version 4.6.
[INFO] [GLX]: Found swap function: glXSwapIntervalEXT.
[INFO] [GLCore]: Vendor: NVIDIA Corporation, Renderer: NVIDIA GeForce RTX 3060 Ti/PCIe/SSE2.
[INFO] [GLCore]: Version: 4.6.0 NVIDIA 525.85.05.
[INFO] [GLCore]: Using resolution 1049x1052.
[INFO] [udev]: Pad #0 (/dev/input/event29) supports force feedback.
[INFO] [udev]: Pad #0 (/dev/input/event29) supports 4 force feedback effects.
[INFO] [Autoconf]: Xbox 360 Controller configured in port 1.
[INFO] [Joypad]: Found joypad driver: "udev".
[INFO] [GLCore]: Loading stock shader.
[INFO] [slang]: Building pass #0 (N/A)
[INFO] [GLCore]: Not using frame history.
[INFO] [GLCore]: Not using framebuffer feedback.
[ERROR] [DBus]: Failed to suspend screensaver via DBus.
[INFO] [X11]: Suspending screensaver (X11, xdg-screensaver).
[INFO] [Video]: Found display server: "x11".
[INFO] [PulseAudio]: Requested 24576 bytes buffer, got 18432.
[INFO] [Display]: Found display driver: "glcore".
[INFO] [Playlist]: Loading history file: "/home/cm/.config/retroarch/content_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/cm/.config/retroarch/content_music_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/cm/.config/retroarch/content_video_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/cm/.config/retroarch/content_image_history.lpl".
[INFO] [Playlist]: Loading favorites file: "/home/cm/.config/retroarch/content_favorites.lpl".
[INFO] [PulseAudio]: Pausing.
```
**After launching**
```
[INFO] [Core]: No content, starting dummy core.
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[INFO] [XINERAMA]: Xinerama version: 1.1.
[INFO] [XINERAMA]: Xinerama screens: 1.
[INFO] [XINERAMA]: Saved monitor #0.
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
[INFO] RetroArch 1.14.0 (Git ad89b0c655)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
[INFO] Capabilities: MMX MMXEXT SSE SSE2 SSE3 SSE4 SSE4.2 AES AVX AVX2
[INFO] Built: Dec 13 2022
[INFO] Version: 1.14.0
[INFO] Git: ad89b0c655
[INFO] =================================================
[INFO] [Input]: Found input driver: "x".
[INFO] [Environ]: SET_VARIABLES.
[INFO] [Environ]: SET_VARIABLES.
[INFO] [Overrides]: Redirecting save file to "/home/cm/.config/retroarch/saves/V4L2.srm".
[INFO] [Overrides]: Redirecting save state to "/home/cm/.config/retroarch/states/V4L2.state".
[ERROR] [Content]: Libretro core requires content, but nothing was provided.
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[INFO] [Core]: Saved core options file to "/home/cm/.config/retroarch/config/V4L2/V4L2.opt".
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Set video size to: 960x720.
[ERROR] [Wayland]: Failed to connect to Wayland server.
[INFO] [GLX]: GLX_EXT_swap_control_tear supported.
[INFO] [GLCore]: Found GL context: "x".
[INFO] [GLCore]: Detecting screen resolution: 1920x1080.
[INFO] [GLX]: X = 0, Y = 0, W = 960, H = 720.
[INFO] [GLX]: Creating context for requested version 3.2.
[INFO] [GLX]: Not running Mesa, trying higher versions...
[INFO] [GLX]: Creating context for version 4.6.
[INFO] [GLX]: Found swap function: glXSwapIntervalEXT.
[INFO] [GLCore]: Vendor: NVIDIA Corporation, Renderer: NVIDIA GeForce RTX 3060 Ti/PCIe/SSE2.
[INFO] [GLCore]: Version: 4.6.0 NVIDIA 525.85.05.
[INFO] [GLCore]: Using resolution 1049x1052.
[INFO] [udev]: Pad #0 (/dev/input/event29) supports force feedback.
[INFO] [udev]: Pad #0 (/dev/input/event29) supports 4 force feedback effects.
[INFO] [Joypad]: Found joypad driver: "udev".
[INFO] [GLCore]: Loading stock shader.
[INFO] [slang]: Building pass #0 (N/A)
[INFO] [GLCore]: Not using frame history.
[INFO] [GLCore]: Not using framebuffer feedback.
[ERROR] [DBus]: Failed to suspend screensaver via DBus.
[INFO] [X11]: Suspending screensaver (X11, xdg-screensaver).
[INFO] [Video]: Found display server: "x11".
[INFO] [PulseAudio]: Requested 24576 bytes buffer, got 18432.
[INFO] [Display]: Found display driver: "glcore".
[INFO] [Playlist]: Loading history file: "/home/cm/.config/retroarch/content_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/cm/.config/retroarch/content_music_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/cm/.config/retroarch/content_video_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/cm/.config/retroarch/content_image_history.lpl".
[INFO] [Playlist]: Loading favorites file: "/home/cm/.config/retroarch/content_favorites.lpl".
[INFO] [PulseAudio]: Pausing.
```
### Relevant configs
**~/.config/retroarch/config/V4L2/V4L2.opt**
```
videoproc_capture_mode = "alternate"
videoproc_frame_times = "Off"
videoproc_output_mode = "progressive"
videoproc_videodev = "/dev/video0"
```

If I understood correctly it seems we are missing `[RETRO_ENVIRONMENT_SET_SUPPORT_NO_GAME ](https://github.com/libretro/RetroArch/blob/3d9c60be7cd7643134827f5e9a0945fe74d25789/libretro-common/include/libretro.h#L662-L668)environment like [libretro-net-retropad](https://github.com/libretro/RetroArch/blob/3d9c60be7cd7643134827f5e9a0945fe74d25789/cores/libretro-net-retropad/net_retropad_core.c#L281) and [libretro-2048](https://github.com/libretro/libretro-2048/blob/ca1a08cbb50128cbf79150f0518f1ffa5e814b6e/libretro.c#L281) in [retro_set_environment](https://github.com/libretro/RetroArch/blob/3d9c60be7cd7643134827f5e9a0945fe74d25789/cores/libretro-video-processor/video_processor_v4l2.c#L271-L303).

But when I added it, I got unfortunately segmentation fault using v42loopback 
```

[DEBUG] [Environ]: GET_VARIABLE: videoproc_videodev = "/dev/video0"
[ERROR] [Environ]: GET_VARIABLE: videoproc_audiodev - Invalid value.
[DEBUG] [Environ]: GET_VARIABLE: videoproc_capture_mode = "alternate"
[DEBUG] [Environ]: GET_VARIABLE: videoproc_output_mode = "progressive"
[DEBUG] [Environ]: GET_VARIABLE: videoproc_frame_times = "Off"
[DEBUG] [Environ]: GET_VARIABLE: videoproc_videodev = "/dev/video0"
[ERROR] [Environ]: GET_VARIABLE: videoproc_audiodev - Invalid value.
/dev/video0:
  Driver: v4l2 loopback
  Card: Virtual
  Bus Info: platform:v4l2loopback-000
  Version: 5.15.90
[DEBUG] [Environ]: GET_VARIABLE: videoproc_videodev = "/dev/video0"
[ERROR] [Environ]: GET_VARIABLE: videoproc_audiodev - Invalid value.
[DEBUG] [Environ]: GET_VARIABLE: videoproc_capture_mode = "alternate"
[DEBUG] [Environ]: GET_VARIABLE: videoproc_output_mode = "progressive"
[DEBUG] [Environ]: GET_VARIABLE: videoproc_frame_times = "Off"
VIDIOC_G_STD failed: Inappropriate ioctl for device
libv4l2: error allocating conversion buffer
VIDIOC_DQBUF failed: Invalid argument
Segmentation fault (core dumped)
```
To be continued ;)

--- a/cores/libretro-video-processor/video_processor_v4l2.c
+++ b/cores/libretro-video-processor/video_processor_v4l2.c
@@ -283,6 +283,9 @@ RETRO_API void VIDEOPROC_CORE_PREFIX(retro_set_environment)(retro_environment_t 
 
    VIDEOPROC_CORE_PREFIX(environment_cb) = cb;
 
+   bool no_content = true;
+   cb(RETRO_ENVIRONMENT_SET_SUPPORT_NO_GAME, &no_content);
+
    /* Allows retroarch to seed the previous values */
    VIDEOPROC_CORE_PREFIX(environment_cb)(RETRO_ENVIRONMENT_SET_VARIABLES, envvars);
 

